### PR TITLE
regional PF clustering for muon 

### PR DESCRIPTION
**Pull request by HuguesBrun**
This PR gives the possibility to do regional PF clustering around muon candidate at HLT. It does not affect the behavior of the current `EgammaHLTRechitInRegionsProducer` and `EgammaHLTCaloTowerProducer`.

### CODE REVIEW

File: RecoEgamma/EgammaHLTProducers/src/HLTRechitInRegionsProducer.cc [hunk 1]
```diff
@@ -89,10 +92,11 @@ void EgammaHLTRechitInRegionsProducer::fillDescriptions(edm::ConfigurationDescri
   desc.add<double>("l1LowerThrIgnoreIsolation", 0.0);
   desc.add<double>("regionEtaMargin", 0.14);
   desc.add<double>("regionPhiMargin", 0.4);
-  descriptions.add(("hltEgammaHLTRechitInRegionsProducer"), desc);  
+  descriptions.add(std::string("hlt")+std::string(typeid(HLTRechitInRegionsProducer<T1>).name()), desc);
```
Comment (Martin-Grunewald) [inline RecoEgamma/EgammaHLTProducers/src/HLTRechitInRegionsProducer.cc]: The above line would be correct in 73X, but not in 74X:
here please adapt:
```
descriptions.add(defaultModuleLabel<HLTRechitInRegionsProducer<T>>(), desc);
```

**Comment by Martin-Grunewald** [comment]
-1

Fix above. Also, make a PR for 73X if you want to use this before HLT moves to 74X.

**Comment by Martin-Grunewald** [comment]
Did you actually test this code?
IE, I get an error:
```
%MSG
02-Feb-2015 07:26:30 CET  Closed file file:RelVal_Raw_FULL_MC.root
----- Begin Fatal Exception 02-Feb-2015 07:26:30 CET-----------------------
An exception of category 'MultiplePlugins' occurred while
   [0] Constructing the EventProcessor
Exception Message:
The plugin 'EgammaHLTRechitInRegionsProducer' is found in multiple files
 '"pluginRecoEgammaEgammaHLTProducers.so"'
 '"pluginRecoEgammaEgammaHLTProducers.so"'
in directory '/scratch/CMS2/CMSSW_7_4_X_2015-02-02-0200/lib/slc6_amd64_gcc481'.
The code must be changed so the plugin only appears in one plugin file. You will need to remove the macro which \
registers the plugin so it only appears in one of these files.
  If none of these files register such a plugin, then the problem originates in a library to which all these fil\
es link.
The plugin registration must be removed from that library since plugins are not allowed in regular libraries.
----- End Fatal Exception -------------------------------------------------
```

**Comment by HuguesBrun** [comment]
Hi Martin,

When I do the change to use ```defaultModuleLabel``` I am not able to compile anymore. 
I added in the ```HLTrigger/HLTcore``` in the Buildfile but I does not help. 
Do you have an idea of what can be the problem ? I do not understand what I am doing wrong respect to the other modules using defaultModuleLabel...
errorscreeshot

**Comment by HuguesBrun** [comment]
The compilation problem is not here anymore if I declare the plugin from the HLTRechitInRegionsProducer.cc

Will backport it in 73X

**Comment by Martin-Grunewald** [comment]
@cmsbuild please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
@nclopezo 
this PR seems stuck in "testing" for 17h - please check

**Comment by nclopezo** [comment]
Hi @Martin-Grunewald,
There was a glitch in Jenkins, I restarted the tests.

**Pull request merged**